hermes-management/src/main/java/pl/allegro/tech/hermes/management/infrastructure/kafka/service/KafkaBrokerTopicManagement.java

```diff
@@ -39,15 +43,21 @@ public KafkaBrokerTopicManagement(TopicProperties topicProperties, AdminClient k
     public void createTopic(Topic topic) {
         Map<String, String> config = createTopicConfig(topic.getRetentionTime().getDurationInMillis(), topicProperties);
 
-        kafkaNamesMapper.toKafkaTopics(topic).forEach(k ->
+        kafkaNamesMapper.toKafkaTopics(topic).stream().map(k ->
                 kafkaAdminClient.createTopics(Collections.singletonList(
                         new NewTopic(
                                 k.name().asString(),
                                 topicProperties.getPartitions(),
                                 (short) topicProperties.getReplicationFactor()
                         ).configs(config)
                 ))
-        );
+        ).forEach(result -> {
+            try {
+                result.all().get();
+            } catch (InterruptedException | ExecutionException e) {
+                throw new BrokersClusterCommunicationException(e);
+            }
+        });
```

Comment:
There are similar problems in a few other places in this class:
* line 65
* line 74
* line 124

Reply:
Thanks for this catch! I've extracted `waitForKafkaFuture` method, which now is applied to all places you mentioned.